### PR TITLE
bugfix: Fix decode path issue.

### PR DESCRIPTION
**Pull request by xiaolanglanglang**
Does not work correctly when the path is Chinese.

/你好/ Couldn't find /你好/index.html correctly.

### CODE REVIEW

**Comment by SukkaW** [comment]
Already fixed in #99 #100 #105.

**Comment by xiaolanglanglang** [comment]
```
  if (ASSET_MANIFEST[rawPathKey]) {
    requestKey = request
  } else if (ASSET_MANIFEST[decodeURIComponent(rawPathKey)]) {
    pathIsEncoded = true;
    requestKey = request
  } else {
    requestKey = options.mapRequestToAsset(request)
  }
```
Example of the file ```tags/你好/index.b1cce8c1f1.html```
request ```http://example.com/tags/你好/```

```
  if (ASSET_MANIFEST['tags/%e4%bd%a0%e5%a5%bd/']) {  //false
    requestKey = request
  } else if (ASSET_MANIFEST[decodeURIComponent('tags/你好/')]) {  //false
    pathIsEncoded = true;
    requestKey = request
  } else {
    requestKey = options.mapRequestToAsset(request)   //tags/%e4%bd%a0%e5%a5%bd/index.html
  }
```
In this case, the requestKey value can't find ```tags/你好/index.b1cce8c1f1.html```.

Your commit seems to solve the filename problem with filenames like ```你好.b1cce8c1f1.html```, but it fails when you need to add ```/index.html```.

**Comment by xiaolanglanglang** [comment]
Do I need to add some clarification?

File: src/index.ts [hunk 1]
```diff
@@ -107,7 +107,14 @@ const getAssetFromKV = async (event: FetchEvent, options?: Partial<Options>): Pr
     pathIsEncoded = true;
     requestKey = request
   } else {
-    requestKey = options.mapRequestToAsset(request)
+    const mapedRequest = mapRequestToAsset(request)
```
Comment (signalnerve) [inline src/index.ts:110]:
```suggestion
    const mappedRequest = mapRequestToAsset(request)
```

Comment (signalnerve) [reply src/index.ts:110]: Should be `mapped` instead of `maped` across the board here - can you update the rest of this code path too?

**Review by signalnerve — CHANGES_REQUESTED** [review]
So sorry for the massive delay in reviewing this, most of the contributors to this project internally aren't here now so I'm picking this project up this week 😬  A few things for you to fix so we can properly review this!

**Review by xiaolanglanglang — COMMENTED** [review]
I have changed it.

**Review by Cherry — APPROVED** [review]
LGTM. Thanks!

**Comment by signalnerve** [comment]
```
360c465 ~/src/kv-asset-handler> npm test

> @cloudflare/kv-asset-handler@0.1.1 pretest /Users/kristianfreeman/src/kv-asset-handler
> npm run build


> @cloudflare/kv-asset-handler@0.1.1 build /Users/kristianfreeman/src/kv-asset-handler
> tsc -d


> @cloudflare/kv-asset-handler@0.1.1 test /Users/kristianfreeman/src/kv-asset-handler
> ava dist/test/*.js --verbose


  ✔ mapRequestToAsset › mapRequestToAsset() correctly changes /about -> /about/index.html
  ✔ mapRequestToAsset › mapRequestToAsset() correctly changes /about/ -> /about/index.html
  ✔ mapRequestToAsset › mapRequestToAsset() correctly changes /about.me/ -> /about.me/index.html
  ✔ serveSinglePageApp › serveSinglePageApp returns root asset path when request path ends in .html
  ✔ serveSinglePageApp › serveSinglePageApp returns root asset path when request path does not have extension
  ✔ serveSinglePageApp › serveSinglePageApp returns requested asset when request path has non-html extension
Environment doesnt support readable streams
Environment doesnt support readable streams
  - getAssetFromKV › getAssetFromKV when body not empty, should invoke .cancel()
  ✔ getAssetFromKV › getAssetFromKV does not cache on Cloudflare when bypass cache set
  ✔ getAssetFromKV › getAssetFromKV if not in asset manifest still returns nohash.txt
  ✔ getAssetFromKV › getAssetFromKV when setting browser caching
  ✔ getAssetFromKV › getAssetFromKV when namespace not bound fails
  ✔ getAssetFromKV › getAssetFromKV return correct val from KV and default caching
  ✔ getAssetFromKV › getAssetFromKV evaluated the file matching the extensionless path first /client/ -> client
  ✔ getAssetFromKV › getAssetFromKV evaluated the file matching the extensionless path first /client -> client
  ✔ getAssetFromKV › getAssetFromKV if no asset manifest /client -> client fails
  ✔ getAssetFromKV › getAssetFromKV if sub/ -> sub/index.html served
  ✔ getAssetFromKV › getAssetFromKV gets index.html by default for / requests
  ✔ getAssetFromKV › getAssetFromKV non ASCII path support
  ✔ getAssetFromKV › getAssetFromKV supports browser percent encoded URLs
  ✔ getAssetFromKV › getAssetFromKV supports user percent encoded URLs
  ✔ getAssetFromKV › getAssetFromKV custom key modifier
  ✔ getAssetFromKV › getAssetFromKV with no trailing slash on root
  ✔ getAssetFromKV › getAssetFromKV with no trailing slash on a subdirectory
  ✔ getAssetFromKV › getAssetFromKV no result throws an error
  ✔ getAssetFromKV › getAssetFromKV passing in a custom NAMESPACE serves correct asset
  ✔ getAssetFromKV › getAssetFromKV when setting custom cache setting
  ✔ getAssetFromKV › getAssetFromKV when custom namespace without the asset should fail
  ✔ getAssetFromKV › getAssetFromKV only decode URL when necessary
  ✔ getAssetFromKV › getAssetFromKV Support for user decode url path
  ✔ getAssetFromKV › getAssetFromKV when etag and if-none-match are present but if-none-match !== etag, should bypass cache
  ✔ getAssetFromKV › getAssetFromKV caches on two sequential requests
  ✔ getAssetFromKV › getAssetFromKV if-none-match not sent but resource in cache, should return hit
  ✔ getAssetFromKV › getAssetFromKV if range request submitted and resource in cache, request fulfilled
  ✔ getAssetFromKV › getAssetFromKV TTls set to null should not cache on browser or edge (115ms)
  ✔ getAssetFromKV › getAssetFromKV does not store max-age on two sequential requests (120ms)
  ✔ getAssetFromKV › getAssetFromKV when if-none-match === etag and etag === pathKey in manifest, should revalidate (112ms)
Environment doesnt support readable streams
Environment doesnt support readable streams
  ─

  35 tests passed
  1 test todo
```

**Pull request merged**